plugins/redis/redis_cli.go

```diff
@@ -14,8 +14,10 @@ func RedisCLI() schema.Executable {
 		DocsURL: sdk.URL("https://redis.io/docs/ui/cli"),
 		NeedsAuth: needsauth.IfAll(
 			needsauth.NotWhenContainsArgs("-u"),
-			needsauth.NotWhenContainsArgs("--user"),
-			needsauth.NotWhenContainsArgs("-a"),
+			needsauth.NotWhenContainsArgs("--user"), // skip when a custom username is provided
+			needsauth.NotWhenContainsArgs("-a"),     // skip when a custom password is provided
+			needsauth.NotWhenContainsArgs("-h"),     // skip when a custom host is provided
+			needsauth.NotWhenContainsArgs("-p"),     // skip when a custom port is provided
```

Comment:
Can you help me understand why we need to do this? Is it to allow a user to connect to a redis instance without a password?

What if the users wants to use credentials in 1Password to connect to a redis instance on a custom host/port? (or if the 1Password item does not have the host/port specified)

Reply:
@arunsathiya Correct me if I'm wrong but I think the desired behaviour is overwriting the host and port a user might have set up in 1Password with the ones that the user is specifying per command basis. All while still using the password stored in 1Password.

In this case, this behaviour shouldn't be handled here in the needsAuth, because this means we will skip even the password provisioning if the user just wants to override a host/port. This should be enforced at the time of provisioning the `-h` and `-p` flags to the command line. We need to detect if these are set already, and if yes, don't override them. If they are not set already in the command line, then provisioned the ones from 1Password.